BEP18.md

```diff
@@ -0,0 +1,123 @@
+| Height         | int64       | height of this snapshot                                                                                                                                                                                                                                                                                                                                          |
+| StateHashes    | []SHA256Sum | hashes of tendermint state chunks                                                                                                                                                                                                                                                                                                                                |
+| AppStateHashes | []SHA256Sum | hashes of app state chunks                                                                                                                                                                                                                                                                                                                                       |
+| BlockHashes    | []SHA256Sum | hashes of the block in this snapshot, currently only one block is synced                                                                                                                                                                                                                                                                                             |
+| NumKeys        | []int64     | number of keys for each sub-store, this sacrifices clear boundary between cosmos and tendermint, saying tendermint knows application DB might be organized by sub-stores.<br><br> |
+
+### 5.4 Snapshot chunk format
+
+#### 5.4.1 App state chunk
+
+ App state chunk includes iavl tree nodes. Usually, each app state chunk takes up to 4MB serialized iavl tree nodes (before snappy compression). 
+ 
+ Iavl tree node bigger than 4MB is split into different incomplete chunks, that's where `Completeness` field effect.
+
+| Field        | Type     | Description                                                                                                                                                                                  |
+|--------------|----------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| StartIdx     | int64    | compare (startIdx and number of complete nodes) against (Manifest.NumKeys) we can know each node's sub-store                                                                                      |
+| Completeness | uint8    | flag of completeness of this chunk, not enum because of go-amino doesn't support enum encoding                                                                                               |
```

Comment:
can't this be a `bool`?

Reply:
> can't this be a `bool`?

It has 4 kind of value: 0 complete, 1 - in complete start, 2 - in complete middle, 3 - in complete end.
I have left more detailed explaination on `StartIdx` field, you can read that:)

Reply:
please add these values into the explanation.